manifests/kustomize/base/metadata/kustomization.yaml

```diff
@@ -12,3 +12,9 @@ resources:
   - metadata-writer-role.yaml
   - metadata-writer-rolebinding.yaml
   - metadata-writer-sa.yaml
+  - execution-cache-deployer-clusterrole.yaml
```

Comment:
this caching component could be its own component. its dependency on metadata is implementation details. Consider having its own folder.